src/test/java/warehouse/CartTest.java

```diff
@@ -35,4 +36,38 @@ void getTotalPrice() {
     @Test
     void addToCartById() {
     }
+
+    @Test
+    void testRemoveToCartById_withOneId() throws InvalidIdException {
+        Cart cart = new Cart();
+        cart.addProductInCart(new Product(DeviceType.TABLET, "ASUS", "Tozzo", "a occhio", 26, 500, 499, 699, 25));
+        cart.removeFromCartById(new FilterById(25));
+        boolean result = cart.isEmpty();
+        assertTrue(result);
+    }
+
+    @Test
+    void testRemoveToCartById_withMultipleId() throws InvalidIdException {
+        Cart cart = new Cart();
+        cart.addProductInCart(new Product(DeviceType.TABLET, "ASUS", "Tozzo", "a occhio", 26, 500, 499, 699, 25));
+        cart.addProductInCart(new Product(DeviceType.SMARTPHONE, "ASUS", "Quadrato", "a occhio", 26, 500, 499, 699, 26));
+        cart.addProductInCart(new Product(DeviceType.NOTEBOOK, "ASUS", "Cubico", "a occhio", 26, 500, 499, 699, 27));
+        cart.removeFromCartById(new FilterById(25));
+        cart.removeFromCartById(new FilterById(26));
+        cart.removeFromCartById(new FilterById(27));
+        boolean result = cart.isEmpty();
+        assertTrue(result);
+    }
+
+    @Test
+    void testRemoveToCartById_withOneId_addedByWarehouse() throws InvalidIdException {
+        Warehouse warehouse = new Warehouse();
+        Cart cart = new Cart();
+        warehouse.addProductToStock(new Product(DeviceType.TABLET, "ASUS", "Tozzo", "a occhio", 26, 500, 499, 699, 25));
+        cart.addToCartById(new FilterById(1), warehouse);
+        cart.removeFromCartById(new FilterById(1));
+        boolean result = cart.isEmpty();
+        assertTrue(result);
+    }
+
```

Comment:
E se il carrello fosse vuoto?

Reply:
In che senso? Il carrello è già vuoto dopo aver utilizzato il metodo

Reply:
se fosse vuoto prima di usare il metodo

Reply:
ah okok verifico